src/Microsoft.Framework.Configuration.Binder/ConfigurationBinder.cs

```diff
@@ -175,6 +194,40 @@ private static void BindCollection(object collection, Type collectionType, IConf
             }
         }
 
+        private static Array BindArray(Array source, IConfiguration config)
+        {
+            var children = config.GetChildren().ToArray();
+            var arrayLength = source.Length;
+            var elementType = source.GetType().GetElementType();
+            var newArray = Array.CreateInstance(elementType, arrayLength + children.Length);
+
+            // binding to array has to preserve already initialized arrays with values
```

Comment:
is very dangerous feature!!! wi bind initialized object and get old values + new values. Expexted result: new values replaces old